### PR TITLE
C++ API parity: Sigmoid

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27538 C++ API parity: Threshold
* #27537 C++ API parity: Tanhshrink
* #27536 C++ API parity: Tanh
* #27535 C++ API parity: Softsign
* #27534 C++ API parity: Softshrink
* #27489 C++ API parity: Softplus
* **#27488 C++ API parity: Sigmoid**

Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

### CODE REVIEW

File: torch/csrc/api/include/torch/nn/functional/activation.h [hunk 1]
```diff
@@ -74,6 +74,10 @@ inline Tensor celu(Tensor& input, const CELUOptions& options = {}) {
   }
 }
 
+inline Tensor sigmoid(const Tensor& input) {
+  return torch::sigmoid(input);
+}
```
Comment (yf225) [inline torch/csrc/api/include/torch/nn/functional/activation.h]: We can probably remove `F::sigmoid`, because the Python version mentions that `nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.`

File: torch/csrc/api/src/nn/modules/activation.cpp [hunk 2]
```diff
@@ -189,5 +189,17 @@ void CELUImpl::pretty_print(std::ostream& stream) const {
   stream << ")";
 }
 
+// ============================================================================
+
+Tensor SigmoidImpl::forward(const Tensor& input) {
+  return sigmoid(input);
```
Comment (yf225) [inline torch/csrc/api/src/nn/modules/activation.cpp]: nit: would be great to write:
```suggestion
  return torch::sigmoid(input);
```
to strictly match Python version :D

**Review by yf225 — APPROVED** [review]
Thanks a lot @pbelevich!
